cmd/prysmctl/withdrawal/withdrawal.go

```diff
@@ -0,0 +1,165 @@
+package withdrawal
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/logrusorgru/aurora"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/rpc/apimiddleware"
+	"github.com/prysmaticlabs/prysm/v3/io/prompt"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	"go.opencensus.io/trace"
+)
+
+const (
+	basePath = "/eth/v1"
+	apiPath  = "/beacon/pool/bls_to_execution_changes"
+)
+
+func setWithdrawalAddress(c *cli.Context, r io.Reader) error {
+	ctx, span := trace.StartSpan(c.Context, "withdrawal.blsToExecutionAddress")
+	defer span.End()
+
+	BeaconNodeHost := "http://localhost:3500"
+	if c.String(BeaconHostFlag.Name) != "" {
+		BeaconNodeHost = c.String(BeaconHostFlag.Name)
+	}
+	u, err := url.ParseRequestURI(BeaconNodeHost)
+	if err != nil {
+		return errors.Wrap(err, "invalid format, unable to parse url")
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return fmt.Errorf("url must be in the format of http(s)://host:port url used: %v", BeaconNodeHost)
+	}
+
+	foundFilePaths, err := findWithdrawalFiles(c.String(FileFlag.Name))
+	if err != nil {
+		return err
+	}
+	if len(foundFilePaths) == 0 {
+		return errors.New("no compatible files were found")
+	}
+	au := aurora.NewAurora(true)
+	fmt.Println(au.Red("===============IMPORTANT==============="))
+	fmt.Println(au.Red("Please read the following carefully"))
+	fmt.Println("This action will allow you to partially withdraw any amount over the 32 staked eth in your validator balance. " +
+		"You will also be entitled to the full withdrawal if your validator has exited. " +
+		"Please navigate to the following website and make sure you understand the current implications " +
+		"of changing your bls withdrawal address to an ethereum address. " +
+		"THIS ACTION WILL NOT BE REVERSIBLE ONCE INCLUDED. " +
+		"You will NOT be able to change the address again once changed. ")
+	for _, foundFilePath := range foundFilePaths {
+		b, err := os.ReadFile(filepath.Clean(foundFilePath))
+		if err != nil {
+			return errors.Wrap(err, "failed to open file")
+		}
+		switch string(b)[0:1] {
+		case "[":
+			var to []*apimiddleware.SignedBLSToExecutionChangeJson
+			if err := json.Unmarshal(b, &to); err != nil {
+				return errors.Wrap(err, "failed to unmarshal file")
+			}
+			if len(to) == 0 {
+				return errors.New("the list of signed requests is empty")
+			}
+			for _, jsonOb := range to {
+				if err := verifyWithdrawalCertainty(r, jsonOb); err != nil {
+					return err
+				}
+			}
+			return callWithdrawalEndpoint(ctx, BeaconNodeHost, to)
```

Comment:
We can call the endpoint only once, with a single list containing objects from all files.

Reply:
will change this to handle that